src/Executor/ExecutionResult.php

```diff
@@ -142,10 +142,15 @@ public function toArray(int $debug = DebugFlag::NONE) : array
                 return array_map($formatter, $errors);
             };
 
-            $result['errors'] = $errorsHandler(
+            $handledErrors = $errorsHandler(
                 $this->errors,
                 FormattedError::prepareFormatter($this->errorFormatter, $debug)
             );
+
+            // While we know that there were errors initially, they might have been discarded
+            if ($handledErrors) {
```

Comment:
I'd evaluate boolean expression here. Guess we want to check for non-emptiness?

```suggestion
            if ($handledErrors !== []) {
```

Reply:
It's the same thing, empty array is falsy

Reply:
Yup but it's not type safe. When type changes it does not get caught. Triggers error with phpstan with strict types extension on.

https://phpstan.org/r/d5fe9f4e-cdb7-49ce-88ed-8f17f85cb691

Reply:
I feel it's unnecessarily verbose and has no real benefit, but if the plan is to turn on phpstan strict, I don't mind

Reply:
At least in my opinion it's better to rely on types so we need less type-related testing in the long run 🤷🏾‍♀️ .